Modules/TRD/src/PulsePositionCheck.cxx

```diff
@@ -0,0 +1,155 @@
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PulsePositionCheck.cxx
+/// \author Deependra (deependra.sharma@cern.ch)
+///
+
+#include "TRD/PulsePositionCheck.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+#include "QualityControl/QcInfoLogger.h"
+// ROOT
+#include <TH1.h>
+#include <TF1.h>
+#include <TMath.h>
+
+#include <fairlogger/Logger.h> 
+#include <DataFormatsQualityControl/FlagReasons.h>
+#include "CCDB/BasicCCDBManager.h"
+
+using namespace std;
+using namespace o2::quality_control;
+
+namespace o2::quality_control_modules::trd
+{
+
+void PulsePositionCheck::configure() 
+{
+  if (auto param = mCustomParameters.find("ccdbtimestamp"); param != mCustomParameters.end()) {
+    mTimeStamp = std::stol(mCustomParameters["ccdbtimestamp"]);
+    ILOG(Debug, Support) << "configure() : using ccdbtimestamp = " << mTimeStamp << ENDM;
+  } else {
+    mTimeStamp = o2::ccdb::getCurrentTimestamp();
+    ILOG(Debug, Support) << "configure() : using default timestam of now = " << mTimeStamp << ENDM;
+  }
+  auto& mgr = o2::ccdb::BasicCCDBManager::instance();
+  mgr.setTimestamp(mTimeStamp);
+  ILOG(Debug, Devel) << "initialize PulseHeight" << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+  if (auto param = mCustomParameters.find("pulseheightpeaklower"); param != mCustomParameters.end()) {
+    mPulseHeightPeakRegion.first = stof(param->second);
+    ILOG(Debug, Support) << "configure() : using pulseheightpeaklower " << mPulseHeightPeakRegion.first << ENDM;
+  } else {
+    mPulseHeightPeakRegion.first = 1.0;
+    ILOG(Debug, Support) << "configure() : using default pulseheightpeaklower  = " << mPulseHeightPeakRegion.first << ENDM;
+  }
+  if (auto param = mCustomParameters.find("pulseheightpeakupper"); param != mCustomParameters.end()) {
+    mPulseHeightPeakRegion.second = stof(param->second);
+    ILOG(Debug, Support) << "configure() : using pulseheightpeakupper = " << mPulseHeightPeakRegion.second << ENDM;
+  } else {
+    mPulseHeightPeakRegion.second = 5.0;
+    ILOG(Debug, Support) << "configure() : using default pulseheightpeakupper = " << mPulseHeightPeakRegion.second << ENDM;
+  }
+}
+
+Quality PulsePositionCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  Quality result = Quality::Null;
+
+  // you can get details about the activity via the object mActivity:
+  ILOG(Debug, Trace) << "Run type: " << getActivity()->mType << ENDM;
+
+  for (auto& [moName, mo] : *moMap) {
+
+    (void)moName;
+    if (mo->getName() == "mPulseHeight") {
+
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
+	    
+      //check max bin is in the spike on left.
+      auto maxbin = h->GetMaximumBin();
+
+      if (maxbin < mPulseHeightPeakRegion.first || maxbin > mPulseHeightPeakRegion.second) {
+        // is the peak in the peak region.
+        result = Quality::Bad;
+        result.addReason(FlagReasonFactory::Invalid(),
```

Comment:
Here is the list of flags you can use:
https://github.com/AliceO2Group/AliceO2/blob/dev/DataFormats/QualityControl/etc/flagReasons.csv
These should normally explain what is the effect on data. If you do not know what is the effect on data, use `Unknown`, but feel free to propose new flags if you think something is missing.

`Invalid` is a special flag and  should not be used by detector code.